src/libstore/gc-store-api.cc

```diff
@@ -0,0 +1,13 @@
+#include "gc-store-api.hh"
+
+namespace nix {
+
+GcStore & requireGcStore(Store & store0)
+{
+    auto * store = dynamic_cast<GcStore *>(&store0);
+    if (!store)
+        throw UsageError("Garbage collection not supported by this store");
+    return *store;
```

Comment:
```suggestion
GcStore & requireGcStore(Store & store)
{
    auto * gcStore = dynamic_cast<GcStore *>(&store);
    if (!gcStore)
        throw UsageError("Garbage collection not supported by this store");
    return *gcStore;
```

and in general use `store` and `gcStore` instead of `store0` and `store`.